Source/WebKit/WebProcess/Network/webrtc/WebMDNSRegister.cpp

```diff
@@ -78,10 +78,11 @@ void WebMDNSRegister::registerMDNSName(ScriptExecutionContextIdentifier identifi
     auto requestIdentifier = MDNSRegisterIdentifier::generate();
     m_pendingRegistrations.add(requestIdentifier, PendingRegistration { WTFMove(callback), identifier, ipAddress });
 
-    // FIXME: Use async reply.
     auto& connection = WebProcess::singleton().ensureNetworkProcessConnection().connection();
-    if (connection.send(Messages::NetworkMDNSRegister::RegisterMDNSName { requestIdentifier, identifier, ipAddress }, 0) != IPC::Error::NoError)
-        finishedRegisteringMDNSName(requestIdentifier, { }, MDNSRegisterError::Internal);
+    connection.sendWithAsyncReply(Messages::NetworkMDNSRegister::RegisterMDNSName { requestIdentifier, identifier, ipAddress }, [weakThis = WeakPtr { *this }] (MDNSRegisterIdentifier&& identifier, String&& mdnsName, std::optional<MDNSRegisterError> error) {
```

Comment:
Does requestIdentifier mean anything anymore, since we have a completion handler?